### DISCUSSION TITLE
Get a list of all available versions of MSBuild

### DISCUSSION THREAD

**Discussion by dimonovdd** [cake-build/cake#4083] [Extension Q&A]
Can I somehow get a list of all installed versions of MSBuild and their location? (As in a screenshot of Rider)

I have a project build set up on 2 macs and 1 windows. And these three devices have a different path to MSBuild. Only 16 version is suitable for building the project

![telegram-cloud-photo-size-2-5231306545357242553-x](https://user-images.githubusercontent.com/59065470/206683784-583d99a8-b441-4af9-ade6-1661e7b7aedf.jpg)

**Comment by dimonovdd** [comment]
OR maybe How can I get a variable from .bash_profile?

**Reply by nils-a** [reply]
You can access environment variables using `EnvironmentVariable(string)` see https://cakebuild.net/dsl/environment/#Environment-Variables

**Comment by nils-a** [comment]
Cake has built-in support for `VsWhere`, see https://cakebuild.net/dsl/vswhere/

**Reply by dimonovdd** [reply]
I tried, it doesn't seem to work on macOS

**Reply by nils-a** [reply]
That might be. You best check with the [VsWhere project](https://github.com/microsoft/vswhere) whether they support non-Windows visual studio versions.
